src/lightning_app/utilities/load_app.py

```diff
@@ -113,6 +115,31 @@ def _create_fake_main_module(script_path):
     return module
 
 
+@contextmanager
+def _patch_sys_argv():
+    from lightning_app.cli.lightning_cli import run_app
+
+    original_argv = sys.argv
+    sys.argv = [v for v in sys.argv if v not in ("lightning", "run", "app")]
+    if "--app_args" not in sys.argv:
+        new_argv = sys.argv[:1]
+    else:
+        options = [p.opts[0] for p in run_app.params[1:] if p.opts[0] != "--app_args"]
+        argv_slice = sys.argv
+        first_index = argv_slice.index("--app_args") + 1
+        matches = [
+            argv_slice.index(opt) for opt in options if opt in argv_slice and argv_slice.index(opt) >= first_index
+        ]
+        if not matches:
+            last_index = len(argv_slice)
+        else:
+            last_index = min(matches)
+        new_argv = [argv_slice[0]] + argv_slice[first_index:last_index]
+    sys.argv = new_argv
+    yield
+    sys.argv = original_argv
```

Comment:
@tchaton - Please ignore this suggestion if you do not agree, but a comment here on what we are doing here (/and why), and if we can use arg parser in the future for this will probably help?

Reply:
+1
comments and docstrings here will be very helpful